wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/Emitter.scala

```diff
@@ -349,7 +366,9 @@ object Emitter {
 
       // See genIdentityHashCode in HelperFunctions
       factory.callMethodStatically(BoxedDoubleClass, hashCodeMethodName),
-      factory.callMethodStatically(BoxedStringClass, hashCodeMethodName)
+      factory.callMethodStatically(BoxedStringClass, hashCodeMethodName),
+      factory.instantiateClass(WasmMemorySegmentClass, WasmMemorySegmentCtor),
+      factory.instantiateClass(WasmMemoryAllocatorClass, WasmMemoryAllocatorCtor)
```

Comment:
This is what causes the tests failure, because the backend now refuses to run at all without these classes.

Ideally, instead, the backend should adapt to whether these classes are available or not. So not declare them as symbol requirements. Instead, only generate the helper functions that manipulate them *if* they exist in the list of `LinkedClass`es we receive.